aiida/orm/nodes/data/code/installed.py

```diff
@@ -72,11 +72,11 @@ def validate_filepath_executable(self):
         try:
             with override_log_level():  # Temporarily suppress noisy logging
                 with self.computer.get_transport() as transport:
-                    file_exists = transport.isfile(self.filepath_executable)
-        except Exception:  # pylint: disable=broad-except
+                    file_exists = transport.isfile(str(self.filepath_executable))
```

Comment:
If `local` returned a `str` here and `ssh` a `Path`, should we then not also aim to harmonize the behavior of the `filepath_executable` property?

Reply:
They both returned a `pathlib.Path`, the problem is that `LocalTransport.isfile` accepts a `pathlib.Path`, whereas `SshTransport` only accepts strings. We should probably update the transport interfaces to add support for `pathlib.Path` objects.

Reply:
Thanks - yes, that would make sense. Up to you whether you want to do it here or not

Reply:
I think it is better to do the whole interface in one go. Have opened a feature request for it #5797